GitDepend/Visitors/CheckArtifactsVisitor.cs

```diff
@@ -0,0 +1,215 @@
+        /// Visits a project dependency.
+        /// </summary>
+        /// <param name="directory">The directory of the project.</param>
+        /// <param name="dependency">The <see cref="Dependency" /> to visit.</param>
+        /// <returns>
+        /// The return code.
+        /// </returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public ReturnCode VisitDependency(string directory, Dependency dependency)
+        {
+            //read in the nuget packages created in the artifacts folder
+            var nugetPackageFiles = _fileSystem.Directory.EnumerateFiles(directory + '/' + dependency.Configuration.Packages.Directory);
+            //get the versionNumbers, they come in the format of {ProjectName}.{version#}.nupkg
+            var directoryLess = nugetPackageFiles.Select(file => Path.GetFileName(file));
+
+            var extensionLess = directoryLess.Select(file => file.Replace(".nupkg", "")).ToList();
+
+            foreach (var file in extensionLess)
+            {
+                string versionNumber, packageName;
+                GetPackageNameAndVersion(file, out versionNumber, out packageName);
+                if (!_dependencyPackageNamesAndVersions.ContainsKey(packageName))
+                {
+                    _dependencyPackageNamesAndVersions.Add(packageName, versionNumber);
+                }
+            }
+
+            if (_dependencyPackageNamesAndVersions.Count == EMPTY)
+            {
+                DependenciesThatNeedBuilding.Add(dependency.Configuration.Name);
+                return ReturnCode.DependencyPackagesNotBuilt;
+            }
+
+            return ReturnCode.Success;
+        }
+        
+        /// <summary>
+        /// Visists a project.
+        /// </summary>
+        /// <param name="directory">The directory of the project.</param>
+        /// <param name="config">The <see cref="GitDependFile" /> with project configuration information.</param>
+        /// <returns>
+        /// The return code.
+        /// </returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public ReturnCode VisitProject(string directory, GitDependFile config)
+        {
+            var packagesFiles = _fileSystem.Directory.GetFiles(directory, "packages.config", SearchOption.AllDirectories);
+
+            //build a dictionary of current packages that need to be checked.
+            var neededPackages = GetPackagesFromPackagesConfigFiles(packagesFiles);
+            //check only the ones that match
+            var misMatching = CheckMatches(neededPackages);
+
+            if (misMatching.Count != 0)
+            {
+                ProjectsThatNeedNugetUpdate.Add(config.Name);
+                return ReturnCode.DependencyPackagesMisMatch;
+            }
+
+            return ReturnCode.Success;
+        }
+        
+        private void GetPackageNameAndVersion(string file, out string versionNumber, out string packageName)
```

Comment:
This might be easier to do with a Regex. There should be an example of how to extract this information in the BuildAndUpdate visitor.